vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -149,42 +176,31 @@ private List<String> getReplicateLabels() {
   public void addRow() {
     int lastRowIndex = getRows() - 1; // zero-based index
     int increasedRowIndex = lastRowIndex + 1;
-
     for (int columnIndex = 0; columnIndex < header.size(); columnIndex++) {
       SamplesheetHeaderName colHeader = header.get(columnIndex);
       switch (colHeader) {
-        case SPECIES -> prefillCell(columnIndex, increasedRowIndex, species);
-        case SPECIMEN -> prefillCell(columnIndex, increasedRowIndex, specimens);
-        case ANALYTE -> prefillCell(columnIndex, increasedRowIndex, analytes);
-        case CONDITION -> prefillCell(columnIndex, increasedRowIndex,
-            conditionsToReplicates.keySet().stream().toList());
+        case ROW -> generateRowHeaderCell(columnIndex, increasedRowIndex);
+        case SPECIES -> generatePrefilledCell(columnIndex, increasedRowIndex, species);
+        case SEQ_ANALYSIS_TYPE ->
+            generatePrefilledCell(columnIndex, increasedRowIndex, analysisTypes);
+        case SAMPLE_LABEL, CUSTOMER_COMMENT ->
+            generatePrefilledCell(columnIndex, increasedRowIndex, new ArrayList<>());
         case BIOLOGICAL_REPLICATE_ID ->
-            prefillCell(columnIndex, increasedRowIndex, getReplicateLabels());
-        case ROW -> fillEnumerationCell(columnIndex, increasedRowIndex);
-        default -> {
-          DropdownColumn column = dropdownCellFactory.getColumn(columnIndex);
-          if (column != null) {
-            column.increaseToRow(increasedRowIndex);
-          }
-        }
+            generatePrefilledCell(columnIndex, increasedRowIndex, getReplicateLabels());
+        case SPECIMEN -> generatePrefilledCell(columnIndex, increasedRowIndex, specimens);
+        case ANALYTE -> generatePrefilledCell(columnIndex, increasedRowIndex, analytes);
+        case CONDITION -> generatePrefilledCell(columnIndex, increasedRowIndex,
+            conditionsToReplicates.keySet().stream().toList());
       }
-      //cells need to be unlocked if they are not prefilled in any way
-      enableEditableCellsOfColumnUntilRow(increasedRowIndex, columnIndex);
     }
     setMaxRows(increasedRowIndex + 1); // 1-based
   }
 
-  private boolean isPrefilledColumn(int columnIndex) {
-    SamplesheetHeaderName colHeader = header.get(columnIndex);
-    switch (colHeader) {
-      case ROW, SEQ_ANALYSIS_TYPE, BIOLOGICAL_REPLICATE_ID, SPECIES, SPECIMEN, ANALYTE, CONDITION:
-        return true;
-      default:
-        return false;
-    }
-  }
-
-  private void fillEnumerationCell(int colIndex, int rowIndex) {
+  /**
+   * Adds rows to the spreadsheet that contains prefilled data, selectable dropdowns and editable
+   * free-text cells. The rows are added below the last row containing data.
+   */
+  private void generateRowHeaderCell(int colIndex, int rowIndex) {
```

Comment:
Are you sure this is what's done here? This method was written to add the numbers (in the left-most column) and I don't see any new features added to it?